pygdf/libgdf_groupby.py

```diff
@@ -222,8 +226,10 @@ def agg(self, args):
         if isinstance(args, (tuple, list)):
             for agg_type in args:
 
-                val_columns_out = [val + '_' +
-                                   agg_type for val in self._val_columns]
+                # we don't need to change the output column names
+#                val_columns_out = [val + '_' +
```

Comment:
We need to think about what we want to do with the output column names, because this as it is wont work. If you have, for example groupby.agg(["min", "max"]) where the val_columns is just ["val"] then the outputs which would be min(val) and max(val)  would both be put in an output called "val", i.e.  result["val"] . So we either make result["val"] be a structure which can contain both (i think this is what numba is doing), or we have to have it with different names, ie. "val_min" and "val_max" like the old group by is currently doing